features/features.go

```diff
@@ -939,4 +939,12 @@ var (
 						enhancementPR("https://github.com/openshift/enhancements/pull/1752").
 						enableIn(configv1.DevPreviewNoUpgrade).
 						mustRegister()
+
+	FeatureGateHyperShiftOnlyDynamicResourceAllocation = newFeatureGate("HyperShiftOnlyDynamicResourceAllocation").
+								reportProblemsToJiraComponent("hypershift").
+								contactPerson("csrwng").
+								productScope(kubernetes).
+								enhancementPR("https://github.com/kubernetes/enhancements/issues/4381").
+								enableForClusterProfile(Hypershift, configv1.Default).
```

Comment:
I think the `configv1.Default` is always accompanied by the other two cluster types. Moreover, I am not sure if this feature gets enabled in the DevPreviewNoUpgrade, TechPreviewNoUpgrade cluster types if not specified here.
```suggestion
								enableForClusterProfile(Hypershift, configv1.Default, configv1.DevPreviewNoUpgrade, configv1.TechPreviewNoUpgrade).
```